daml-script/daml3/Daml/Script/Questions/Submit/Error.daml

```diff
@@ -22,20 +23,23 @@ data DevErrorType
 
 -- | Additional debugging info optionally provided by the ledger
 data ContractNotFoundAdditionalInfo
-  = NotFound with -- The generic "this contract does not exist"
-      templateIdentifier : Text
+  = NotFound -- The generic "this contract does not exist"
   | NotActive with -- The contract did exist, but has since been archived
-      templateIdentifier : Text
+      additionalInfoCid : AnyContractId
   | NotEffective with -- The contract "will" exist, only thrown from the IDELedger where one can go backwards in time
-      templateIdentifier : Text
+      additionalInfoCid : AnyContractId
       effectiveAt : Text
   | NotVisible with -- The contract exists, but the readAs/actAs party union cannot see it
-      templateIdentifier : Text
+      additionalInfoCid : AnyContractId
       actAs : [Party]
       readAs : [Party]
       observers : [Party]
   deriving Show
 
+isNotActive : ContractNotFoundAdditionalInfo -> Optional AnyContractId
+isNotActive (NotActive {additionalInfoCid}) = Some additionalInfoCid
+isNotActive _ = None
+
```

Comment:
Should we not have one of these per constructor? maybe its easier to stagger the variant into a variant of explicit records to make passing this data back nicer?

Reply:
I'd prefer not even to have these query functions at all anyways. I might resort to using `show` instead and doing the disambiguation in the test expectation